cal/cal/static/js/services.js

```diff
@@ -343,4 +344,58 @@ analyticsApp.service('QueryService', function() {
 
     return [slope, intercept, rSquare];
   }
-});
+
+  this.getDataForCalendarIds = function(type, start, end, calendarId, cache, cacheKey) {
+ //    As an example, tags are saved like this:
+ //    [
+ //     cacheKey1:
+ //       {
+ //         tagId1: {totalHours: 5, label: school, keywords: “UC Berkeley”},
+ //         tagId2: {totalHours: 5, label: music, tag.keywords: “Brandon Flowers”}
+ //       },
+ //     cacheKey2:
+ //       {
+ //         tagId1: {totalHours: 5, label: school, keywords: “UC Berkeley”},
+ // 	      tagId2: {totalHours: 5, label: music, keywords: “Brandon Flowers”}
+ // 	    }
+ //     ]
+
+    if (cache[cacheKey]) {
+      return cache[cacheKey];
+    } else {
+      return $http({
+        method: 'GET',
+        url: '/v1/' + type + '.json',
+        cache: true,
+        params: {
+          start: (start)? start.toISOString() : null,
+          end: (end)? end.toISOString() : null,
+          calendar_ids: JSON.stringify([calendarId]),
+          timezone: moment.tz.guess()
+        }
+      }).then(function successCallback(response) {
+        var modelData = response.data.results;
+        cache[cacheKey] = {};
+        for (var i = 0; i < modelData.length; i++) {
+          var model = modelData[i];
+          if (cache[cacheKey].hasOwnProperty(model.id.toString())) {
+            var newHours = cache[cacheKey][model.id][hours] + model.hours;
+            cache[cacheKey][model.id][hours] = newHours;
```

Comment:
I also believe you can use dot notation instead of [] to access fields. 
So `cache[cacheKey][model.id].hours`? Just seems clearer that this is an attribute field, not a key value. I was going to say do this for model.id too, but I think it's clearer to user [] because each model.id points to an object.